src/singledocparser.cpp

```diff
@@ -46,6 +46,9 @@ void SingleDocParser::HandleDocument(EventHandler& eventHandler) {
 }
 
 void SingleDocParser::HandleNode(EventHandler& eventHandler) {
+  if (depth > depth_limit) {
+    throw ParserException(m_scanner.mark(), ErrorMsg::BAD_FILE);
```

Comment:
> Is that a serious issue? At this point - the file should be considered garbage anyways and we just need to abort gracefully.

Maybe for most use cases. Arguably for most realistic use cases. But I thought it worth mentioning; if you have deeply nested types, then YAML is _probably_ not the right data structure to use.

> That's pretty neat - didn't think of using destructors to take care of this... But then that's a huge change - at that point, I would rather switch to a non-recursive design, as you said, but that's also a large change.

RAII (and, lifetime guarantees) is wonderful :)